diffyscan/diffyscan.py

```diff
@@ -3,39 +3,106 @@
 import time
 import argparse
 import os
-import subprocess
-import tempfile
 
-from .utils.common import load_config, load_env
-from .utils.constants import DIFFS_DIR, START_TIME, DEFAULT_CONFIG_PATH
-from .utils.explorer import get_contract_from_explorer
-from .utils.github import get_file_from_github, get_file_from_github_recursive, resolve_dep
+from .utils.common import load_config, load_env, prettify_solidity
+
+from .utils.constants import (
+    DIFFS_DIR,
+    DEFAULT_CONFIG_PATH,
+    START_TIME,
+)
+from .utils.explorer import (
+    get_contract_from_explorer,
+    compile_contract_from_explorer,
+    parse_compiled_contract,
+)
+from .utils.github import (
+    get_file_from_github,
+    get_file_from_github_recursive,
+    resolve_dep,
+)
 from .utils.helpers import create_dirs
 from .utils.logger import logger
-
+from .utils.binary_verifier import match_bytecode
+from .utils.hardhat import hardhat
+from .utils.node_handler import get_bytecode_from_node, get_account, deploy_contract
+from .utils.calldata import get_calldata
+from .utils.custom_exceptions import ExceptionHandler, BaseCustomException
 
 __version__ = "0.0.0"
 
 
 g_skip_user_input: bool = False
 
-def prettify_solidity(solidity_contract_content: str):
-    github_file_name = os.path.join(tempfile.gettempdir(), "9B91E897-EA51-4FCC-8DAF-FCFF135A6963.sol")
-    with open(github_file_name, "w") as fp:
-        fp.write(solidity_contract_content)
-    prettier_return_code = subprocess.call(
-        ["npx", "prettier", "--plugin=prettier-plugin-solidity", "--write", github_file_name],
-        stdout=subprocess.DEVNULL)
-    if prettier_return_code != 0:
-        logger.error("Prettier/npx subprocess failed (see the error above)")
-        sys.exit()
-    with open(github_file_name, "r") as fp:
-        return fp.read()
-
-
-def run_diff(config, name, address, explorer_api_token, github_api_token, recursive_parsing=False, prettify=False):
+
+def run_bytecode_diff(
+    contract_address_from_config,
+    contract_name_from_config,
+    contract_source_code,
+    config,
+    deployer_account,
+    local_rpc_url,
+    remote_rpc_url,
+):
+    address_name = f"{contract_address_from_config} : {contract_name_from_config}"
+    logger.divider()
+    logger.info(f"Binary bytecode comparison started for {address_name}")
+    target_compiled_contract = compile_contract_from_explorer(contract_source_code)
+
+    contract_creation_code, deployed_bytecode, immutables = parse_compiled_contract(
+        target_compiled_contract
+    )
+
+    remote_deployed_bytecode = get_bytecode_from_node(
+        contract_address_from_config, remote_rpc_url
+    )
+
+    is_fully_matched = match_bytecode(
+        deployed_bytecode,
+        remote_deployed_bytecode,
+        immutables,
+    )
+
+    if is_fully_matched:
+        return
+
+    logger.info(
+        f"Trying to check bytecode via using calldata and deploying into local node"
+    )
```

Comment:
Shall say here that automated match hasn't worked out

Reply:
done